src/tests/class_tests/openms/source/IdentificationData_test.cpp

```diff
@@ -0,0 +1,386 @@
+{
+  IdentificationData::ParentMolecule protein("");
+  // can't register a parent molecule without accession:
+  TEST_EXCEPTION(Exception::IllegalArgument,
+                 data.registerParentMolecule(protein));
+  TEST_EQUAL(data.getParentMolecules().empty(), true);
+
+  protein.accession = "protein_1";
+  protein_ref = data.registerParentMolecule(protein);
+  TEST_EQUAL(data.getParentMolecules().size(), 1);
+  TEST_EQUAL(*protein_ref == protein, true); // "TEST_EQUAL(*parent_ref, parent)" doesn't compile
+  IdentificationData::ParentMolecule rna("rna_1",
+                                         IdentificationData::MoleculeType::RNA);
+  rna_ref = data.registerParentMolecule(rna);
+  TEST_EQUAL(data.getParentMolecules().size(), 2);
+  TEST_EQUAL(*rna_ref == rna, true); // "TEST_EQUAL(*parent_ref, parent)" doesn't compile
+}
+END_SECTION
+
+START_SECTION((const ParentMoleculeGroupings& getParentMoleculeGroupings() const))
+{
+  TEST_EQUAL(data.getParentMoleculeGroupings().empty(), true);
+  // tested further below
+}
+END_SECTION
+
+START_SECTION((void registerParentMoleculeGrouping(const ParentMoleculeGrouping& grouping)))
+{
+  IdentificationData::ParentMoleculeGroup group;
+  group.parent_molecule_refs.insert(protein_ref);
+  group.parent_molecule_refs.insert(rna_ref);
+  IdentificationData::ParentMoleculeGrouping grouping;
+  grouping.label = "test_grouping";
+  grouping.groups.insert(group);
+  data.registerParentMoleculeGrouping(grouping);
+  TEST_EQUAL(data.getParentMoleculeGroupings().size(), 1);
+  TEST_EQUAL(data.getParentMoleculeGroupings()[0].groups.size(), 1);
+  TEST_EQUAL(data.getParentMoleculeGroupings()[0].groups.begin()->parent_molecule_refs.size(), 2);
+}
+END_SECTION
+
+START_SECTION((const IdentifiedPeptides& getIdentifiedPeptides() const))
+{
+  TEST_EQUAL(data.getIdentifiedPeptides().empty(), true);
+  // tested further below
+}
+END_SECTION
+
+START_SECTION((IdentifiedPeptideRef registerIdentifiedPeptide(const IdentifiedPeptide& peptide)))
+{
+  IdentificationData::IdentifiedPeptide peptide(AASequence::fromString(""));
+  // can't register a peptide without a sequence:
+  TEST_EXCEPTION(Exception::IllegalArgument,
+                 data.registerIdentifiedPeptide(peptide));
+  TEST_EQUAL(data.getIdentifiedPeptides().empty(), true);
+
+  // peptide without protein reference:
+  peptide.sequence = AASequence::fromString("TEST");
+  peptide_ref = data.registerIdentifiedPeptide(peptide);
+  TEST_EQUAL(data.getIdentifiedPeptides().size(), 1);
+  TEST_EQUAL(*peptide_ref == peptide, true); // "TEST_EQUAL(*peptide_ref, peptide)" doesn't compile
+
+  // peptide with protein reference:
+  peptide.sequence = AASequence::fromString("PEPTIDE");
+  peptide.parent_matches[protein_ref];
+  peptide_ref = data.registerIdentifiedPeptide(peptide);
+  TEST_EQUAL(data.getIdentifiedPeptides().size(), 2);
+  TEST_EQUAL(*peptide_ref == peptide, true); // "TEST_EQUAL(*peptide_ref, peptide)" doesn't compile
```

Comment:
what happens when you register the same peptide, once with and without protein reference? does it generate 2 entries?